packages/cli-kit/src/private/node/ui/components/TextInput.tsx

```diff
@@ -45,27 +47,39 @@ const TextInput: FunctionComponent<TextInputProps> = ({
     return chalk.inverse(value[0]) + chalk.dim(value.slice(1))
   }
 
+  const cursorChar = figures.square
+  const defaultCursor = (
+    <Text color={color} backgroundColor={color}>
+      {cursorChar}
+    </Text>
+  )
+
   const renderedPlaceholder =
     defaultValue.length > 0
       ? renderPlaceholder(defaultValue)
       : placeholder.length > 0
       ? renderPlaceholder(placeholder)
-      : chalk.inverse(' ')
+      : defaultCursor
 
   // render cursor
   renderedValue = value
     .split('')
     .map((char, index) => {
       if (index === cursorOffset) {
-        return chalk.inverse(char)
+        return shouldDisplayColors() ? chalk.inverse(char) : cursorChar
```

Comment:
Is it possible to add a test case for this behavior with no colors?

Reply:
🤔

Reply:
I haven't had much luck with stubbing `shouldDisplayColors`, there is too much weirdness in vitest module stubbing. Instead, I added an optional `noColor` boolean parameter, which matches [something I've done in the past](https://github.com/Shopify/cli/blob/b2ce9ffed4890b8999c702be7119cf1ac94107ec/packages/cli-kit/src/private/node/ui/components/Scrollbar.tsx#L21) for testing purposes as well.